### PR TITLE
guide: organize ToC a bit

### PR DESCRIPTION
**Pull request by tjhance**
move chapters around and add major headings to break it up a bit

### CODE REVIEW

**Comment by parno** [comment]
This looks like a nice improvement.  While we're at it, can you move the "[INTERLUDE: using assert and assume to develop proofs](develop_proofs.md)" bit to be in the "Understanding the prover" section?

**Comment by Chris-Hawblitzel** [comment]
> This looks like a nice improvement. While we're at it, can you move the "[INTERLUDE: using assert and assume to develop proofs](develop_proofs.md)" bit to be in the "Understanding the prover" section?

It might work well being the very first subsection of "Understanding the prover" (and dropping the word "INTERLUDE").

**Comment by tjhance** [comment]
I considered it, given what we discussed yesterday, but the content of that chapter relates to the previous chapter. I thought it would require a bit more thought.

**Comment by tjhance** [comment]
ok since both chris & bryan suggested it, I moved the chapter and adjusted its introductory text.

File: source/docs/guide/src/develop_proofs.md [hunk 1]
```diff
@@ -1,6 +1,6 @@
 # Using assert and assume to develop proofs
 
-The previous section started with an outline of proof:
+In [an earlier chapter](./spec_lib.md), we started with an outline of proof:
```
Comment (parno) [inline source/docs/guide/src/develop_proofs.md:3]: I think we want "of [a] proof" (I know that was there before you started, but I just noticed it now).

**Pull request merged**